packages/select/src/Select.js

```diff
@@ -241,7 +241,7 @@ const Select = ({
     if (allowSelectAll && attributes.isMulti) {
       if (
         [...options, ...newOptions].length > 0 &&
-        (values[name] === undefined ||
+        (!values[name] ||
```

Comment:
I don't think this should come up often, but it's possible to have 
```js
const values = {
  a: true,
  b: false,
  c: null,
  d: '',
}
```
and then 
`!values[a] = false`
`!values[b] = true` -> false positive
`!values[c] = true` -> desired behavior for `null`
`!values[d] = true` -> potential false positive?
`!values[e] = true` -> desired behavior for `undefined`

when really it looks like we only want `true` returned for `undefined` or `null` values. So I think we should just make the check explicit for those two values

Reply:
Good catch. This is fixed